### PR TITLE
chore(export): use FileInput for export output selection

### PR DESCRIPTION
**Pull request by addaleax**
[First commit is to avoid conflicts with #3693, the second commit undoes it anyway]

Allow usage of `FileInput` with the electron file open/save API,
so that we can also use it in situations in which we want to
write to output files and still have a consistent UI with the
readonly/input case, and use it in the export modal.

After #3693:

Screenshot from 2022-11-03 15-12-38

After this PR (without the other changes from #3693):

Screenshot from 2022-11-03 15-09-29

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-import-export/src/components/export-select-output/export-select-output.tsx [hunk 1]
```diff
@@ -1,14 +1,12 @@
 import { SelectFileType } from '../select-file-type';
-import { IconTextButton } from 'hadron-react-buttons';
 import toNS from 'mongodb-ns';
-import fileSaveDialog from '../../utils/file-save-dialog';
 import ProgressBar from '../progress-bar';
 import { FILETYPE } from '../../constants/export-step';
+// @ts-expect-error No types for .less?
```
Comment (gribnoysup) [inline packages/compass-import-export/src/components/export-select-output/export-select-output.tsx:5]: You can add something like this to typings.d.ts for this package to be handled semi-correctly:

https://github.com/mongodb-js/compass/blob/ea4f317e964c0504e54c01d83c385280a9135bba/packages/compass-aggregations/src/typings.d.ts#L8-L11

Comment (addaleax) [reply packages/compass-import-export/src/components/export-select-output/export-select-output.tsx]: Thank you! Done :)

**Review by Anemy — COMMENTED** [review]
Changes look good, nice tests too. I like the separating of the backend for down the line vscode or web usage. The ui/ux makes more sense than the current selection imo as well 👍 

One question, and one edge case we probably want to cover:
Question: Looks like we now show an error when a user chooses a file in the import modal, then reopens the file explorer and closes it, should we handle that error silently instead?:
Screen Shot 2022-11-03 at 12 43 31 PM

Edge case: With these changes when choosing a file in the connection form, then removing that file, it's not possible to choose the recently removed file and have it show up in the form:

https://user-images.githubusercontent.com/1791149/199782358-79ca5a3c-5547-4ddd-8bc7-88f40359cfd3.mp4

File: packages/compass-components/src/components/file-input.tsx [hunk 2]
```diff
@@ -235,15 +254,20 @@ function FileInput({
           style={{ display: 'none' }}
           // Force a re-render when the values change so
           // the component is controlled by the prop.
-          key={values ? values.join(',') : 'empty'}
```
Comment (Anemy) [inline packages/compass-components/src/components/file-input.tsx:238]: Looks like this key was needed for making the input behave as expected when no file is chosen, or when the file is removed then re-chosen (see video in other comment).

Comment (addaleax) [reply packages/compass-components/src/components/file-input.tsx:238]: @Anemy Yeah, thanks for the pointer! Updated this to bring `key` back :+1:
